### PR TITLE
Move More CI into Jenkins - Travis Migration Progress

### PR DESCRIPTION
**Pull request by bearbin**
Build clang debug and GCC release on Jenkins rather than travis.

Recommend merge soon due to the withdrawal of travis services. Mac builds will be added in due course.

### CODE REVIEW

**Comment by 12xx12** [comment]
I also liked the ARM build. That found some possible mistakes

File: Jenkinsfile [hunk 1]
```diff
@@ -27,9 +27,35 @@ pipeline {
...
+                }
+                stage("clang") {
+                    environment {
+                        TRAVIS_CUBERITE_BUILD_TYPE = 'Debug'
+                        TRAVIS_JOB_NUMBER = "{$env.BUILD_ID}"
+                        CC = "clang"
+                        CXX = "clang++"
+                    }
+                    steps {
+                        sh 'bash ./travisbuild.sh'
```
Comment (12xx12) [inline Jenkinsfile:50]: Is this inteded to have the gcc only build release and clang only debug? On travis we had every combination

**Comment by tigerw** [comment]
Various versions of the compiler and cmake too, but with the untimely demise of ~~Bob the Builder~~ Travis this is a must-merge unless we go without CI. Thanks bearbin for migrating!

**Comment by 12xx12** [comment]
I actually don't understand what changed in the transition from travis-ci.org to travis-ci.com. Can anyone explain? There are still "free" open source offers. Or are they limited in a way that this is not suitable for us`?

**Comment by bearbin** [comment]
It's free, but in the sense that you have to ask them nicely every month, and next month they might say "no, no builds for you any more". Better to migrate to something we control, that won't suddenly disappear.

File: Jenkinsfile (hunk 1 above)
Comment (bearbin) [reply Jenkinsfile:50]: Yes, for now.

**Comment by bearbin** [comment]
I would like to get a full matrix of builds up and running, but for now we only have so much builder space and the builds can't run for a year per commit.

I'm working on getting some auto-scaling cloud builders working, but because of other committments that will have to be a next week thing.

**Comment by 12xx12** [comment]
> 
> 
> It's free, but in the sense that you have to ask them nicely every month, and next month they might say "no, no builds for you any more". Better to migrate to something we control, that won't suddenly disappear.

I see. Thanks

**Comment by mathiascode** [comment]
https://daniel.haxx.se/blog/2021/06/14/bye-bye-travis-ci/ sums up a few concerns.

**Comment by 12xx12** [comment]
> 
> 
> I would like to get a full matrix of builds up and running, but for now we only have so much builder space and the builds can't run for a year per commit.
> 
> I'm working on getting some auto-scaling cloud builders working, but because of other committments that will have to be a next week thing.

Next week is very fast :D Dont worry

**Review by 12xx12 — APPROVED** [review]

**Pull request merged**